client/ayon_core/pipeline/farm/pyblish_functions.py

```diff
@@ -495,33 +491,49 @@ def _get_real_frames_to_render(frames):
     return frames_to_render
 
 
-def _get_real_files_to_rendered(collection, frames_to_render):
-    """Use expected files based on real frames_to_render.
+def _get_real_files_to_render(collection, frames_to_render):
+    """Filter files with frames that should be really rendered.
+
+    'expected_files' are collected from DCC based on timeline setting. This is
+    being calculated differently in each DCC. Filtering here is on single place
+
+    But artists might explicitly set frames they want to render in Publisher UI
+    This range would override and filter previously prepared expected files
+    from DCC.
 
-    Artists might explicitly set frames they want to render via Publisher UI.
-    This uses this value to filter out files
     Args:
-        frames_to_render (list): of str '1001'
-    """
-    files = [os.path.basename(f) for f in list(collection)]
-    file_name, extracted_frame = list(collect_frames(files).items())[0]
+        collection (clique.Collection): absolute paths
+        frames_to_render (list[int]): of int 1001
+    Returns:
+        (list[str])
 
-    if not extracted_frame:
-        return files
+    Example:
+    --------
 
-    found_frame_pattern_length = len(extracted_frame)
+    expectedFiles = [
+        "foo_v01.0001.exr",
+        "foo_v01.0002.exr",
+    ]
+    frames_to_render = '0001'
+    >>
+    ["foo_v01.0001.exr"] - only explicitly requested frame returned
+    """
+    found_frame_pattern_length = collection.padding
```

Comment:
Do note that `clique.assemble` creates a `clique.Collection` where `padding` is `0` if it has no frame starting with `0` by default. See the `assume_padded_when_ambiguous` argument to `clique.assemble` in recent `clique` versions [here](https://clique.readthedocs.io/en/2.0.0/api_reference/index.html#clique.assemble).

The question is however whether that could be problematic. If e.g. someone has explicit frame to render 999 but the assembled sequence was 1001-1010 then padding would be zero instead of four meaning we'd get:
- 999.exr
- 1000.exr
- 1001.exr 

Instead of:
- 0999.exr
- 1000.exr
- 1001.exr

Whether that's problematic, I'm not sure - just wanted to clarify that.

Reply:
Also, wouldn't it just work to do:

```python

# Get overlapping frame ranges
included_frames = set(collection.indexes).intersection(frames_to_render)

# Create matching collection that refers to those frames
# and get their filenames.
collection = 
collection.indexes.clear()
collection.indexes.update(included_frames)
return list(collection)
```

Note that this changes the input `collection` - if you don't want that you may need to create a copy collection or just do:
```python
included_frames = set(collection.indexes).intersection(frames_to_render)
real_collection = clique.Collection(
    collection.head,
    collection.tail,
    collection.padding,
    indexes=included_frames
)
return list(real_collection)
```